### PR TITLE
feat: complete test cases of cluster, proxy and agent contracts

### PR DESCRIPTION
**Pull request by ashuralyk**
# Description
we lack test cases to cover the transfer and burn operations for cluster, proxy and agent contracts

# Related Issue
- [ ] #31
- [x] #35 
- [x] #38

### CODE REVIEW

File: lib/utils/src/mime.rs
Comment (ashuralyk) [inline lib/utils/src/mime.rs:1]: is this enough to achieve MIT protocol requirements? @quake

File: lib/utils/src/mime.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// note: impelemtation refers to https://github.com/hyperium/mime
```
Comment (ashuralyk) [inline lib/utils/src/mime.rs:1]: is this enough to achieve MIT protocol requirements? @quake

File: contracts/cluster_agent/src/entry.rs [hunk 2]
```diff
@@ -111,7 +111,7 @@ fn process_transfer() -> Result<(), Error> {
     let input_agent_data = load_cell_data(0, GroupInput)?;
     let output_agent_data = load_cell_data(0, GroupOutput)?;
 
-    if input_agent_data != output_agent_data {
+    if input_agent_data != output_agent_data || input_agent_data.is_empty() {
```
Comment (Flouse) [inline contracts/cluster_agent/src/entry.rs:114]: https://github.com/sporeprotocol/spore-contract/blob/feat/complete-test-cases/contracts/cluster_agent/src/entry.rs#L47-L50

input_agent_data should be proxy_type_hash, a.k.a Hash(ClusterProxyCell.Type).
So, it won't be empty.

Comment (ashuralyk) [reply contracts/cluster_agent/src/entry.rs:114]: yes, completely, I treat it as a double check

Comment (Flouse) [reply contracts/cluster_agent/src/entry.rs:114]: LGTM

**Review by Flouse — APPROVED** [review]

File: tests/src/tests/spore.rs [hunk 3]
```diff
@@ -102,291 +141,401 @@ fn test_multi_spores_mint() {
...
+            &spore_out_point,
+            spore_type_id.to_vec().into(),
         );
+        let spore_input = build_spore_input(&mut context, type_.clone(), serialized.clone());
+
+        let normal_output = build_normal_output(&mut context);
+        let tx = TransactionBuilder::default()
+            .input(spore_input)
+            .output(normal_output)
+            .output_data(serialized.as_slice().pack())
```
Comment (Flouse) [inline tests/src/tests/spore.rs:517]:
```suggestion
            .output_data(Default::default())
```

destory the spore data too.

**Pull request merged**